src/Uno.Extensions.Templates/content/unoapp-extensions/.template.config/template.json

```diff
@@ -459,29 +700,76 @@
           }
         },
         {
-          "condition": "(!markup)",
+          "condition": "(!use-csharp-markup)",
           "exclude": [
-            "MyExtensionsApp.UI/Extensions/ApplicationBuilderExtensions.cs",
-            "MyExtensionsApp.UI/**/Styles/*.cs"
+            "MyExtensionsApp/Infrastructure/ApplicationBuilderExtensions.cs",
+            "MyExtensionsApp/Styles/*.cs"
           ]
         },
         {
-          "condition": "(markup)",
+          "condition": "(use-csharp-markup)",
           "exclude": [
-            "MyExtensionsApp.UI/**/Styles/*.xaml",
-            "MyExtensionsApp.UI/**/Views/*.xaml"
+            "MyExtensionsApp/Styles/*.xaml",
+            "MyExtensionsApp/**/*.xaml"
           ],
           "rename": {
-            "MyExtensionsApp.UI/Views/MainPage.xaml.cs": "MyExtensionsApp.UI/Views/MainPage.cs",
-            "MyExtensionsApp.UI/Views/SecondPage.xaml.cs": "MyExtensionsApp.UI/Views/SecondPage.cs",
-            "MyExtensionsApp.UI/Views/Shell.xaml.cs": "MyExtensionsApp.UI/Views/Shell.cs"
+            "MyExtensionsApp/MainPage.xaml.cs": "MyExtensionsApp/MainPage.cs",
+            "MyExtensionsApp/Presentation/MainPage.xaml.cs": "MyExtensionsApp/Presentation/MainPage.cs",
+            "MyExtensionsApp/Presentation/SecondPage.xaml.cs": "MyExtensionsApp/Presentation/SecondPage.cs",
+            "MyExtensionsApp/Presentation/Shell.xaml.cs": "MyExtensionsApp/Presentation/Shell.cs"
           }
+        },
+        {
+          "condition": "(blank-app-template)",
+          "rename": {
+            "MyExtensionsApp.Base/App.blank.cs": "MyExtensionsApp.Base/App.xaml.cs"
+          },
+          "exclude": [
+            "MyExtensionsApp/Business/**",
+            "MyExtensionsApp/Infrastructure/**",
+            "MyExtensionsApp/Presentation/**",
+            "MyExtensionsApp/Services/**",
+            "MyExtensionsApp/Styles/**",
+            "MyExtensionsApp/appsettings.*",
+            "MyExtensionsApp.Base/App.default.cs",
+            "MyExtensionsApp.Tests/AppInfoTests.cs"
+          ]
+        },
+        {
+          "condition": "(default-app-template)",
+          "rename": {
+            "MyExtensionsApp.Base/App.default.cs": "MyExtensionsApp.Base/App.xaml.cs"
+          },
+          "exclude": [
+            "MyExtensionsApp.Base/App.blank.cs",
+            "MyExtensionsApp.Tests/UnitTest1.cs"
+          ]
+        },
+        {
+          "condition": "(blank-app-template || navigation == 'frame')",
+          "exclude": [
+            "MyExtensionsApp/Presentation/**"
+          ]
+        },
+        {
+          "condition": "(default-app-template && navigation != 'frame')",
+          "exclude": [
+            "MyExtensionsApp/MainPage.xaml",
+            "MyExtensionsApp/MainPage.xaml.cs"
+          ]
+        },
+        {
+          "condition": "(tfm == 'net7.0')",
+          "exclude": [
+            "MyExtensionsApp.DataContracts/_Compat/**"
```

Comment:
See earlier comment about using net6.0 - we should be able to remove the _Compact folder completely